### PR TITLE
Instructions for restoring a bacpac file

### PR DESCRIPTION
**Pull request by CarlSargunar**
This is an amend to the documentation, with thanks to Seb for his help. It outlines the steps required to restore a bacpac file from an Umbraco Cloud backup to a local SQL Server instance

### CODE REVIEW

**Comment by CarlSargunar** [comment]
This fixes issue  #4484

**Comment by CarlSargunar** [comment]
Note - on the failing check, the database IS called master, that's an unfortunate historical artefact of SQL server

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 1]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
 - It is very important that you enter the database name. If you do not, the connection will fail.
 - Expand "Databases", right-click your database (it should be the only one listed), select "Tasks", then select "Export Data-tier Application...".
 - Proceed through the dialog, setting the options appropriate to your situation, to save the "bacpac" file. This is your database backup.
+
+
+## Restoring a cloud backup to a SQL Server Database
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:27]:
```suggestion
## Restoring a Cloud backup to a SQL Server Database
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 2]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
...
+
+In order to restore a .bacpac file to a SQL Server database, you will need to first ensure that the SQL Server has has 'Contained Database Authentication' enabled. This is a security feature that is disabled by default, and without enabling it you will not be able to restore the database. For refrence please see [this article](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:29]:
```suggestion
To restore a `.bacpac` file to a SQL Server database, you need to ensure that the SQL Server has **Contained Database Authentication** enabled. This is a security feature that is disabled by default. Without enabling it you will not be able to restore the database. For reference, check out the [Microsoft documentation on the topic](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 3]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
...
+
+To Enable Contained Database Authentication, run the following SQL against your sql server on the Master database
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:31]:
```suggestion
To Enable Contained Database Authentication, run the following SQL against your SQL server on the Master database.
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 4]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
...
+
+    sp_configure 'contained database authentication', 1;  
+    GO  
+    RECONFIGURE;  
+    GO  
+
+Once this is done you will be able to restore the database using the following steps:
+- Connect to your SQL Server using Sql Server Management Studio (SSMS).
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:39]:
```suggestion
- Connect to your SQL Server using SQL Server Management Studio (SSMS).
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 5]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
...
+- Expand "Databases", right-click "Databases", select "Tasks", then select "Import Data-tier Application...".
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:40]:
```suggestion
- Expand "Databases" and right-click "Databases".
- Select "Tasks" and then "Import Data-tier Application...".
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 6]
```diff
@@ -22,3 +22,22 @@ Follow these steps:
...
+- Proceed through the dialog, by browsing to the saved location of your bacpac file, and then setting the options appropriate to your coniguration
```
Comment (sofietoft) [inline Umbraco-Cloud/Databases/Backups/index.md:41]:
```suggestion
- Proceed through the dialog, by browsing to the saved location of your `.bacpac` file.
- Set the options appropriate to your configuration.
```

**Review by sofietoft — COMMENTED** [review]
Thanks for the PR @CarlSargunar ! 💪 

This is a great addition to the article!
I'll make sure we have the process reviewed as soon as possible.
Also, I made a few suggestions - mainly to grammar and stuff!

**Review by jonat123 — COMMENTED** [review]
@CarlSargunar Thanks for the PR.

I have been testing it out, however, I have had no issues restoring a backup of a cloud site to my local instance with both a database for a v8 site and a database for v10.

On my end, it works just fine without Enableling Contained Database Authentication.

Just to outline what I did:

- I created a project with some content on the cloud (on both v8 and 10)

- I connected to my DB and created a backup by following the docs

- Then I connect to my local instance of SSMS and click on databases and import data-tier application

- I run through the wizard

- And it restores it successfully

Is there something that I am missing?

It seems to me that it should work without enabling the Contained Database Authentication.

**Comment by CarlSargunar** [comment]
@jonat123 it depends if you already had that configuration enabled on your server. Depending on how you have installed SQL Server, and depending on what edition you install that may not be enabled. For example - I normally run a minimal SQL Server through Docker to keep the footprint low, where it was not installed. However if you run the full Dev edition of SQL server locally then that setting may have already been enabled. 

The problem is it's not obvious what the problem is when you try and restore a database. Perhaps the wording should be changed so that we say "If you have a problem, then this is what you need to do". 

For reference, I had a bug which I raised which illustrates the error

https://our.umbraco.com/forum/using-umbraco-and-getting-started//110228-export-bacpac-and-import-into-sql-server

**Comment by jonat123** [comment]
@CarlSargunar Thanks for elaborating  👍

Makes sense to me, as you say the wording should be changed a bit, what do you say to something like this:

> You might need to enable Contained Database Authentication before being able to restore your Umbraco Cloud Database locally.
> 
> 
> To enable it, run the following SQL query against your SQL server on the Master database:
> 
> sp_configure 'contained database authentication', 1;  
> GO  
> RECONFIGURE;  
> GO  

I also think it would make sense if the above is in a note as well :)

**Comment by CarlSargunar** [comment]
cool - I'll do that update in a bit :-)

**Comment by CarlSargunar** [comment]
Updated wording as discussed @jonat123

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 7]
```diff
@@ -22,3 +22,29 @@ Follow these steps:
...
+## Restoring a Cloud backup to a SQL Server Database
+
+Use the following steps:
+- Connect to your SQL Server using Sql Server Management Studio (SSMS).
+- Expand "Databases", right-click "Databases", select "Tasks", then select "Import Data-tier Application...".
+- Proceed through the dialog, by browsing to the saved location of your bacpac file, and then setting the options appropriate to your coniguration
+- Complete the import dialog and the database will be restored.
+
+:::note
+When you restore a bacpac to your SQL server if it fails, check that you have the configuration flag for 'Contained Database Authentication' set to true. If this is not set the import will fail.
```
Comment (jonat123) [inline Umbraco-Cloud/Databases/Backups/index.md:36]:
```suggestion
When restoring a bacpac to your SQL server if it fails, check that you have the configuration flag for 'Contained Database Authentication' set to true. 

If it is not set the import will fail.
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 8]
```diff
@@ -22,3 +22,29 @@ Follow these steps:
...
+:::
+
+To Enable Contained Database Authentication, run the following SQL against your SQL server on the Master database.
+
+    sp_configure 'contained database authentication', 1;  
+    GO  
+    RECONFIGURE;  
+    GO  
+    
+For refrence please see [this article](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```
Comment (jonat123) [inline Umbraco-Cloud/Databases/Backups/index.md:46]:
```suggestion
For reference please see the [contained database authentication documentation](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```

**Review by jonat123 — CHANGES_REQUESTED** [review]
Great, looks good to me @CarlSargunar :) 

Just made a couple of suggestions as it was complaining about the length of the sentence :) 

Also, make sure to have a look at the suggestions from Sofie as well 😄

Once they have been approved we can merge in the PR 😁

**Comment by CarlSargunar** [comment]
Darned sentence length :-)

No worries. I re-structured the PR so @sofietoft I think your suggestions were on the old structure, if you want any more tweaks now please let me know. I've taken any amends which I could take

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 9]
```diff
@@ -22,3 +22,31 @@ Follow these steps:
...
+## Restoring a Cloud backup to a SQL Server Database
+
+Use the following steps:
+- Connect to your SQL Server using Sql Server Management Studio (SSMS).
+- Expand "Databases", right-click "Databases", select "Tasks", then select "Import Data-tier Application...".
+- Proceed through the dialog, by browsing to the saved location of your bacpac file, and then setting the options appropriate to your coniguration
+- Complete the import dialog and the database will be restored.
+
+:::note
+When restoring a bacpac to your SQL server if it fails, check that you have the configuration flag for 'Contained Database Authentication' set to true.
```
Comment (jonat123) [inline Umbraco-Cloud/Databases/Backups/index.md:36]:
```suggestion
When restoring a `bacpac` to your SQL server if it fails, check that you have the configuration flag for 'Contained Database Authentication' set to true. 
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 10]
```diff
@@ -22,3 +22,31 @@ Follow these steps:
...
 - Expand "Databases", right-click your database (it should be the only one listed), select "Tasks", then select "Export Data-tier Application...".
 - Proceed through the dialog, setting the options appropriate to your situation, to save the "bacpac" file. This is your database backup.
+
+
+## Restoring a Cloud backup to a SQL Server Database
+
+Use the following steps:
+- Connect to your SQL Server using Sql Server Management Studio (SSMS).
+- Expand "Databases", right-click "Databases", select "Tasks", then select "Import Data-tier Application...".
+- Proceed through the dialog, by browsing to the saved location of your bacpac file, and then setting the options appropriate to your coniguration
```
Comment (jonat123) [inline Umbraco-Cloud/Databases/Backups/index.md:32]:
```suggestion
- Proceed through the dialog, by browsing to the saved location of your `bacpac` file, and then setting the options appropriate to your configuration
```

File: Umbraco-Cloud/Databases/Backups/index.md [hunk 11]
```diff
@@ -22,3 +22,31 @@ Follow these steps:
...
+:::
+
+To Enable Contained Database Authentication, run the following SQL against your SQL server on the Master database.
+
+    sp_configure 'contained database authentication', 1;  
+    GO  
+    RECONFIGURE;  
+    GO  
+    
+For reference please see the [contained database authentication Server Configuration Option documentation](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```
Comment (jonat123) [inline Umbraco-Cloud/Databases/Backups/index.md:48]:
```suggestion
For reference please see the [Microsoft documentation on the topic](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/contained-database-authentication-server-configuration-option?view=sql-server-ver16).
```

**Comment by jonat123** [comment]
@CarlSargunar Great!

Just read it over one last time and noticed a couple of things and made a couple of suggestions 😄

Once they have been committed I will merge the PR 😁

**Comment by CarlSargunar** [comment]
Wow - I read it too, but totally missed that - shows how much we automatically correct. Thanks for that

**Comment by jonat123** [comment]
@CarlSargunar hehe :D 

Anyways great addition to the documentation :D 

Thanks for the PR, I will merge it in :D

**Pull request merged**